### PR TITLE
Create-end-to-end-deployment

### LINKED ISSUE #319: Improve documentation with e2e examples

### PR DESCRIPTION
**Pull request by Namaskar-1F64F**
closes #319 

This PR adds two main things:
1. A deployment script that steps through the main path of the bond. There is also time to place auction bids as this was necessary while testing the gnosis safe.
2. An ability to add multiple "beneficiaries" to receive, for testing:
    - collateral tokens
    - payment tokens
    - bonds

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -22,4 +34,4 @@ GAS_REPORTER_COINMARKETCAP_API_KEY=ABC123AB-C123A-BC123-ABC1-23ABC123ABC1
 
 ; ---- debug-options ----
 ; Enable gas logging during tests and deployment
-LOG_GAS_USAGE=true
```
[the remark this answers is not in the document]
Comment (Namaskar-1F64F) [reply .env.example:25]: It clogs up the console

File: deploy/deploy_collateral_token.ts [hunk 2]
```diff
@@ -17,18 +21,54 @@ module.exports = async function ({
...
+    tokenDeploymentArguments.symbol,
+    tokenDeploymentArguments.mintAmount,
+    tokenDeploymentArguments.decimals,
+  ];
+  const { address } = await deploy("CollateralToken", {
     contract: "TestERC20",
     from: deployer,
     log: true,
     autoMine: true,
+    waitConfirmations: 1,
```
[the remark this answers is not in the document]
Comment (Namaskar-1F64F) [reply deploy/deploy_collateral_token.ts:36]: guarantees the thing is deployed. important with rinkeby testing, and etherscan verify sometimes fails citing it couldn't be found

File: deploy/deploy_collateral_token.ts [hunk 3]
```diff
@@ -17,18 +21,54 @@ module.exports = async function ({
...
+  const collateralToken = (await ethers.getContractAt(
+    "TestERC20",
+    address
+  )) as TestERC20;
+
+  if (process.env.DEPLOYMENT_BENEFICIARIES) {
+    const beneficiaries = process.env.DEPLOYMENT_BENEFICIARIES.split(",");
+    for (const beneficiary of beneficiaries) {
+      if ((await collateralToken.balanceOf(beneficiary)).gt(0)) {
+        continue;
```
[the remark this answers is not in the document]
Comment (Namaskar-1F64F) [reply deploy/deploy_collateral_token.ts:54]: I like the minimized complexity of an early exit vs potentially an `else` statement for the other condition. This way you know for sure: if there is a balance, do nothing in the loop.

[the remark this answers is not in the document]
Comment (RusseII) [reply deploy/deploy_collateral_token.ts:54]: Hmm. I still like it better without the continue personally. Could use variable names to improve clarity of what is happening here as well. I wouldn't change anything atm tho

**Pull request merged**